ReactWindows/ReactNative.Shared/UIManager/BorderedViewParentManager.cs

```diff
@@ -88,11 +111,15 @@ public void SetBackgroundColor(Border view, uint color)
         /// <param name="view">The view panel.</param>
         /// <param name="color">The color hex code.</param>
         [ReactProp("borderColor", CustomType = "Color")]
-        public void SetBorderColor(Border view, uint? color)
+        public void SetBorderColor(TFrameworkElement view, uint? color)
         {
-            view.BorderBrush = color.HasValue
-                ? new SolidColorBrush(ColorHelpers.Parse(color.Value))
-                : s_defaultBorderBrush;
+            if (HasBorder(view) || color.HasValue && !ColorHelpers.IsTransparent(color.Value))
```

Comment:
This optimization needs to be done more carefully. If we first set border color to 0 (a fully transparent black color), this `if` will ignore the request because the border wouldn't be visible. However if later we set border width to 2, the border will be created with the default border brush, which is an opaque black color, while it should be transparent.